src/components/Search/Search.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { useState } from 'react';
+
+import './Search.css';
+
+type Props = {
+  handleNewSearch(search: string): void;
+  handleError(message: string): void;
+  resetSearch: void;
+};
+
+const Search: React.FC<Props> = ({
+  handleNewSearch,
+  handleError,
+  resetSearch
+}) => {
+
+  const [keyword, setKeyword] = useState('');
+  const [hasResults, setHasResults] = useState(false);
+  const [filter, setFilter] = useState('');
+  const [sort, setSort] = useState('');
+  const [error, setError] = useState('')
+
+  const popularLanguages = [
+    'JavaScript', 'Python', 'C', 'Java', 'Go', 'Perl', 'Ruby', 'Swift', 'Scala', 'PHP', 'C++',
+    'R', 'Objective-C', 'SQL', 'MATLAB', 'Rust', 'TypeScript', 'Kotlin', 'CSS', 'HTML', 'Groovy',
+    'Dart', 'Assembly', 'PowerShell', 'Julia', 'Scratch', 'COBOL', 'Fortran', 'ABAP', 'Scheme', 'Shell',
+    'Prolog', 'VBScript', 'Haskell', 'Delphi', 'Hack', 'Pascal', 'Ada', 'Lua', 'Visual Basic', 'Common Lisp',
+    'Bash', 'Clojure', 'MQL4', 'Apex', 'LabVIEW', 'ABL', 'D', 'SAS', 'Logo', 'C#', 'Jupyter Notebook', 'CoffeeScript',
+    'QML', 'Less', 'Makefile', 'Lua', 'Roff', 'Vala', 'Solidity', 'Starlark', 'TeX', 'Vue', 'React', 'React Native',
+    'Erlang', 'Elixer', 'OCaml', 'OpenSCAD', 'Vim script'
+  ]
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    if(keyword === "") {
+      setHasResults(false)
+      resetSearch()
+      handleError('Enter A Keyword to Start Hunting!')
+      return
+    }
+    handleError('loading...')
+    let search = `${keyword}`
+    if(filter !== '') search += ` language:${filter.toLowerCase()}`
+    if(sort === 'Stars') search += `&sort=stars`
+    handleNewSearch(search)
+    setHasResults(true)
+    handleError('')
+  }
+
+  const handleChange = (e) => {
+    e.preventDefault()
+    if(hasResults) {
+      setHasResults(false)
+      resetSearch()
+    }
+    if(keyword === "") {
+      resetSearch()
+    }
+    setKeyword(e.target.value)
+  }
+
+  const displayingResults = () => {
+    return keyword !== "" && hasResults
+  }
+
+  const buildDropdownOptions = () => {
+    return popularLanguages.map(lang => {
+      return (
+        <option>{lang}</option>
```

Comment:
need a key

Reply:
No error message that I can see. Maybe a key isn't needed for this one. Keep an eye on it

Reply:
Nevermind! Finally errored! Fixed